### PR TITLE
#110 add `Iterable<E>.startsWith(Iterable<E>)` and test

### PR DESCRIPTION
**Pull request by komape**
Added the method `Iterable<E>.startsWith(Iterable<E>)`. This was requested by #110.

### CODE REVIEW

**Comment by komape** [comment]
@leisim Is there anything preventing you from merging this PR? If so, let me know so I can fix it.

File: test/iterable_test.dart [hunk 1]
```diff
@@ -951,5 +951,17 @@ void main() {
...
+    test('.startsWith()', () {
+      expect([].startsWith([]), true);
+      expect([1, 2, 3].startsWith([]), true);
+      expect([1, 2, 3].startsWith([1]), true);
+      expect([1, 2, 3].startsWith([1, 2]), true);
+      expect([1, 2, 3].startsWith([1, 2, 3]), true);
+      expect([1, 2, 3].startsWith([1, 2, 3, 4]), false);
+      expect([1, 2, 3].startsWith([2, 3, 4]), false);
+      expect([1, 2, 3].startsWith([2, 3]), false);
+      expect([1, 2, 3].startsWith([2]), false);
```
Comment (passsy) [inline test/iterable_test.dart:964]: please add a test with `null` values. There might be some tricky corner cases.

Also add a test where `this` is shorter than `otherIterable`

Comment (komape) [reply test/iterable_test.dart:964]: will do the `null` test.

the test in line 961 contains a shorter list for `this` than for the `otherIterable`. or do you mean something else?

Comment (komape) [reply test/iterable_test.dart:964]: passing `null` values to the method is not possible as the method does not accept `nullable` values. a `!` can be appended but this throws an exception before the method is invoked.

```dart
const List<int>? nullList = null;
expect([1, 2, 3].startsWith(nullList!), null);
```
```bash
Null check operator used on a null value
main.<fn>.<fn>
✖ IterableX .startsWith()
Exited (1)
```

 `this` can be null with a `null-aware` operator. in case of a `?` the method does not even get invoked and `null` is directly returned. in case of a `!` an exception is thrown and the method does not get invoked either. I don't see the necessity to test this. 

```dart
const List<int>? nullList = null;
expect(nullList?.startsWith([1, 2, 3]), null);
expect(nullList!.startsWith([1, 2, 3]), null);
```
```bash
Null check operator used on a null value
main.<fn>.<fn>
✖ IterableX .startsWith()
Exited (1)
```

Comment (komape) [reply test/iterable_test.dart:964]: @passsy please check my comments. if i missed something or similar, please let me know.

Comment (passsy) [reply test/iterable_test.dart:964]: Please add the following casees
- `expect([].startsWith([1, 2, 3]), isFalse);` and 
- `expect([null, 1, 2].startsWith([null, 1]), isFalse) `

Comment (komape) [reply test/iterable_test.dart:964]: Oh, you mean `null` elements in the list and not the list itself. Sorry, I should have thought about this.

I don't see the reason why the method should return false if both have a `null` value at the same index. I would argue that the method should return true because both values are equal and it's in the user's responsibility to check for `null` elements. What do you think?

Comment (passsy) [reply test/iterable_test.dart:964]: My fault, of course it should return `true` (corrected it).

Comment (komape) [reply test/iterable_test.dart:964]: added the tests.

**Comment by passsy** [comment]
Fantastic, thanks for your contribution
